### PR TITLE
Qwen3-Omni video/audio joint decoding

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

This PR plugs in video decoding and audio-in-video features for the Qwen3-Omni model:
- **Isolated Multimodal Interfaces**: Separated image and video paths all the way from the CLI wrapper down to the model forward pass (added distinct `videos`/`encoder_videos` and `video_masks`/`encoder_video_masks` arguments).
- **Discrete Mask-based Merging**: In `Decoder._apply_embedding` (src/maxtext/layers/decoders.py), images and videos are now merged into the prompt sequence using separate masks (`bidirectional_mask` vs `bidirectional_mask_video`), enabling clean interleaving of multimodal tokens.
- **Audio-In-Video Preprocessor Offset Bug Fix**: Fixed a bug in `get_mm_offsets_qwen3_omni` (src/maxtext/multimodal/processor_qwen3_omni.py) where audio-in-video offsets were underestimated by 3 tokens (missing structural `<|audio_start|>`/`<|audio_end|>` tokens and incorrectly subtracting for `<|audio_pad|>`). 
- **Padded Waveform JAX Reshape Fix**: Modified `pre_process_audio_qwen3_omni` to pad loaded waveforms to a multiple of 1 second (16,000 samples) prior to feature extraction. This guarantees JAX-friendly log-mel spectrogram multiples of 100 frames, preventing PJRT dynamic reshaping crashes.
- **Reshaped Video preprocessor Output**: Standardized `video_processed` values into 5D `(1, channels, temporal_patches, height_patches, width_patches)` to match the visual encoder's input layer.
- **Unit Test Updated**: 
  - Aligned `tests/unit/qwen3_omni_layers_test.py` with these improvements. The test now pads reference audios and flattens 5D visual arrays prior to comparison. Preprocessing results match HuggingFace with tolerance.
  - With transformers upgrade 4.57.0 -> 5.8.0, the torch_encoder's return interface changed from a tuple of two to a dictionary of three (`last_hidden_state(new) , pooler_output, deepstack_features`). We update the interface in unit tests accordingly and raise the tolerance.

# Tests

```
# Command
python -m maxtext.inference.decode maxtext/configs/base.yml model_name=qwen3-omni-30b-a3b tokenizer_path=Qwen/Qwen3-Omni-30B-A3B-Instruct tokenizer_type=huggingface load_parameters_path=gs://hengtaoguo-maxtext-logs/checkpoints/qwen3-omni-30b-a3b/unscanned/001/0/items per_device_batch_size=1 run_name=ht_test steps=1 async_checkpointing=false scan_layers=false use_multimodal=true use_audio_in_video=True prompt=\'What\ can\ you\ see\ and\ hear\?\ Answer\ in\ two\ short\ sentences.\' video_path=\'tests/assets/test_video.mp4\' max_prefill_predict_length=1250 max_target_length=1280 attention=\'dot_product\' ici_tensor_parallelism=4 add_bos=False hf_access_token=<your_token>

# Results
Input `<|im_start|>user
<|vision_start|><|video_pad|><|vision_end|>What can you see and hear? Answer in two short sentences.<|im_end|>
<|im_start|>assistant
` -> `A large, roaring Tyrannosaurus rex statue is displayed in a museum. The deep, guttural roar of the dinosaur is heard as it appears`
```

```
python -m pytest tests/unit/qwen3_omni_layers_test.py -vv -s
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/inference/maxengine/maxengine.py [hunk 1]
```diff
@@ -418,6 +418,8 @@ def _prefill_jit(
       mrope_deltas: jax.Array | None = None,
       images: jax.Array | None = None,
       image_masks: jax.Array | None = None,
+      videos: jax.Array | None = None,
```
Comment (entrpn) [inline src/maxtext/inference/maxengine/maxengine.py:411]: is this an actual video or something else?

**Review by aireenmei — APPROVED** [review]

File: src/maxtext/inference/maxengine/maxengine.py (hunk 1 above)
Comment (hengtaoguo) [reply src/maxtext/inference/maxengine/maxengine.py:411]: This is an actual video (preprocessed video signal) for decoding. The test example in the description loads a test_video.mp4 asset and could describe the content correctly.

**Review by entrpn — APPROVED** [review]
